OnboardProject/App.tsx

```diff
@@ -1,118 +1,138 @@
-/**
- * Sample React Native App
- * https://github.com/facebook/react-native
- *
- * Generated with the TypeScript template
- * https://github.com/react-native-community/react-native-template-typescript
- *
- * @format
- */
 
 import React from 'react';
+import {useState} from 'react';
+
 import {
   SafeAreaView,
   StyleSheet,
   ScrollView,
   View,
   Text,
   StatusBar,
+  TextInput,
+  Button,
+  TouchableOpacity,
 } from 'react-native';
 
-import {
-  Header,
-  LearnMoreLinks,
-  Colors,
-  DebugInstructions,
-  ReloadInstructions,
-} from 'react-native/Libraries/NewAppScreen';
-
 declare const global: {HermesInternal: null | {}};
 
-const App = () => {
+const App : React.FC = () => {
+
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [hasEmailError, setHasEmailError] = useState(false);
+  const [hasPasswordError, setHasPasswordError] = useState(false);
+
+  function handleLogin(){
+    const emailOk : Boolean = checkEmailFormat();
+    const passwordOk : Boolean = checkPasswordFormat();
+
+    if(emailOk && passwordOk){
+      console.log("Both fields good");
+    }
+    else{
+      console.log("Problem in fields.");
+    }
+  }
+
+  function checkPasswordFormat(){
+
+    if(password.length < 7){
+      setErrorMessage("Password too short");
+      setHasPasswordError(true);
+      return false;
+    }
+
+    const re = new RegExp('^(?:[0-9]+[a-z]|[a-z]+[0-9])[a-z0-9]*$')
```

Comment:
Daora regex!
Essa parte verifica se tem número e letra, certo?
Faria assim 
```suggestion
    const re = /^(?=.*[a-z])(?=.*[0-9]).*$/i
```
Essa parte`(?= )` é um Positive Lookahead assim da pra separa em partes e fica mais fácil de analisar.
Desse jeito fica sem impedir o uso de outros caracteres também